### ISSUE TITLE
Cannot edit a grid cell of an enum column

### ISSUE THREAD

**Issue by ryancwalsh** [dbeaver/dbeaver#21167]
### Description

In DBeaver, when I try to edit a grid cell of an enum column, I get:

```
Reason:
SQL Error [42804]: ERROR: column "flag" is of type "Flag" but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 33
```

My Postgres migration (generated by Prisma) said:
`CREATE TYPE "Flag" AS ENUM ('BANNED', 'FAVORITED');`

So I've tried using 'BANNED' or BANNED in the cell, but they both throw errors. I've also tried saving as 1 in case enums get converted to integers.

https://github.com/dbeaver/dbeaver/issues/72 might be related but isn't exactly what I'm talking about. In my case, I don't need to edit the column definition. I'm trying to edit a cell value in an existing row.

### DBeaver Version

community 23.2.0

### Operating System

Win 10

### Database and driver

Postgres 15

**Comment by ryancwalsh** [comment]
I also tried saving as 'B' or B, but those didn't work.

Writing out a query manually like this works:

```sql
update "public"."Image"
set "flag" = 'BANNED'
where id = 1
```

But it would be nice if cells were editable.

**Comment by LonwoLonwo** [comment]
Hello @ryancwalsh 

Could you also provide a DDL of the "public"."Image" table?

**Comment by ryancwalsh** [comment]
Prisma schema:

```
enum Flag {
  BANNED
  FAVORITED
}

model Image {
  id        Int      @id @default(autoincrement())
  path      String   @unique
  flag      Flag?
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```

became this Postgresql migration:

```sql
CREATE TYPE "Flag" AS ENUM ('BANNED', 'FAVORITED');

CREATE TABLE "Image" (
    "id" SERIAL NOT NULL,
    "path" TEXT NOT NULL,
    "flag" "Flag",
    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
    "updatedAt" TIMESTAMP(3) NOT NULL,

    CONSTRAINT "Image_pkey" PRIMARY KEY ("id")
);

CREATE UNIQUE INDEX "Image_path_key" ON "Image"("path");
```

**Comment by LonwoLonwo** [comment]
Could you please provide footage showing your issue? 

Because I can not reproduce it for now.

Also, related ticket: https://github.com/dbeaver/dbeaver/issues/8104

**Comment by ryancwalsh** [comment]
@LonwoLonwo I opened DBeaver today and started recording a video for you, but this time the enum worked! So I'll close this ticket.

If I ever have the problem again, I'll record a video and re-open the issue. Thanks.

**Comment by hsuabina** [comment]
I found this issue today. For some reason, the table columns tab was not showing my enum-typed column. I got the same error mentioned by OP. Invalidating the DB connection didn't work either

I had to close and reopen DBeaver and then the table was showing the right columns and I could edit a cell (in my case, to do an insert) and it showed the dropdown with all the enum values for me to choose from.

I don't want to reopen this issue, but just confirm that reopening DBeaver seems to fix this issue.
